src/metrics.jl

```diff
@@ -132,7 +132,7 @@ hamming(a::AbstractArray, b::AbstractArray) = evaluate(Hamming(), a, b)
 hamming{T <: Number}(a::T, b::T) = evaluate(Hamming(), a, b)
 
 # Cosine dist
-function eval_start{T<:AbstractFloat}(::CosineDist, a::AbstractArray{T}, b::AbstractArray{T})
+function eval_start{T<:Number}(::CosineDist, a::AbstractArray{T}, b::AbstractArray{T})
```

Comment:
Should probably be `Real` instead of `Number`. I'm not sure that the cosine distance is actually meaningful for complex numbers.